toqito/channels/tests/test_pauli_channel.py

```diff
@@ -0,0 +1,161 @@
+"""Test pauli channel."""
+
+import re
+
+import numpy as np
+import pytest
+from scipy.linalg import eigvalsh
+
+from toqito.channels import partial_trace, pauli_channel
+
+
+@pytest.mark.parametrize(
+    "in_p",
+    [
+        # Negative probability.
+        ([-0.1, 1.1]),
+        # Incorrect length.
+        ([0.2, 0.2, 0.2, 0.2, 0.2]),
+        # Out of range probabilities.
+        ([1.1, 0.1]),
+    ],
+)
+def test_pauli_channel_valid_probability(in_p):
+    """Test probability vector validation with various invalid inputs."""
+    with pytest.raises(
+        ValueError,
+        match=(
+            re.escape("Probabilities must be non-negative and sum to 1.")
+            + "|"
+            + re.escape("The length of the probability vector must be 4^Q for some integer Q (number of qubits).")
+        ),
+    ):
+        pauli_channel(prob=in_p)
+
+
+@pytest.mark.parametrize(
+    "Q,expected_dim",
+    [
+        # Single qubit.
+        (1, 4),
+        # Two qubits.
+        (2, 16),
+        # Three qubits.
+        (3, 64),
+    ],
+)
+def test_pauli_channel_dimensions(Q, expected_dim):

```

Comment:
[nit]: I think where possible and within reason, we want to avoid variable names like `Q` and `p`, especially for input arguments.